service/src/main/resources/migrations/v016__backfila.sql

```diff
@@ -0,0 +1,4 @@
+ALTER TABLE services
+  ADD COLUMN variant VARBINARY(100) NULL DEFAULT NULL,
+  DROP KEY `unq_registry_name`,
+  ADD UNIQUE KEY `unq_registry_name_variant` (registry_name, variant);
```

Comment:
when this is null the unique key does nothing

Reply:
Good call. Do you have a preference for how to approach this?

Off the top of my head, we could:
1. Require a backfill of the column with a default value on the next Backfila upgrade
2. Workaround using [generated columns](https://dev.mysql.com/doc/refman/5.7/en/create-table-generated-columns.html). This places a requirement that Backfila must be on MySQL 5.7.
3. Take a lock at the top of the transaction and rely on the logic in code to enforce the constraint.

Something else?

Reply:
If you want to make "default" the default value, I think you can specify the default value when adding the column and not require a backfill

Reply:
Oh, good point. That seems easiest

Reply:
Ok, this is non-nullable with a default value now